io/jvm/src/main/scala/fs2/io/net/SocketPlatform.scala

```diff
@@ -32,51 +32,11 @@ import java.net.InetSocketAddress
 import java.nio.channels.{AsynchronousSocketChannel, CompletionHandler}
 import java.nio.{Buffer, ByteBuffer}
 
-/** Provides the ability to read/write from a TCP socket in the effect `F`.
-  */
-trait Socket[F[_]] {
-
-  /** Reads up to `maxBytes` from the peer.
-    *
-    * Returns `None` if the "end of stream" is reached, indicating there will be no more bytes sent.
-    */
-  def read(maxBytes: Int): F[Option[Chunk[Byte]]]
-
-  /** Reads exactly `numBytes` from the peer in a single chunk.
-    *
-    * Returns a chunk with size < `numBytes` upon reaching the end of the stream.
-    */
-  def readN(numBytes: Int): F[Chunk[Byte]]
-
-  /** Reads bytes from the socket as a stream. */
-  def reads: Stream[F, Byte]
-
-  /** Indicates that this channel will not read more data. Causes `End-Of-Stream` be signalled to `available`. */
+private[net] trait SocketPlatform[F[_]] {
   def endOfInput: F[Unit]
```

Comment:
I stared at Node's `Socket` API for a very long time and couldn't find it.

https://nodejs.org/api/net.html#net_class_net_socket

This raises an interesting issue: for unsupported methods like this, should we
1. make them platform-specific
2. raise an exception
3. do nothing

(1) Causes the most pain (i.e. full cross) for what could otherwise be pure crosses downstream. But, it makes authors aware of the missing method at compile-time. (2) and (3) still allow for a pure cross, but then there may be unhappy surprises at runtime.

As a motivating example, consider this seemingly innocuous `ember-client` code:
https://github.com/http4s/http4s/blob/main/ember-client/src/main/scala/org/http4s/ember/client/EmberConnection.scala#L30

Reply:
Actually, I think I'm all for (1). Still a bit painful, but downstreams can mitigate (1) by adding implicit syntax extensions to fill in the gaps however they like. Then, their code can purely cross-build.